core/services/pipeline/task_params.go

```diff
@@ -232,13 +232,14 @@ type BytesParam []byte
 func (b *BytesParam) UnmarshalPipelineParam(val interface{}) error {
 	switch v := val.(type) {
 	case string:
+		// try hex first
 		if len(v) >= 2 && v[:2] == "0x" {
 			bs, err := hex.DecodeString(v[2:])
-			if err != nil {
-				return err
+			if err == nil {
+				*b = BytesParam(bs)
+				return nil
 			}
-			*b = BytesParam(bs)
-			return nil
+			// The base64 encoding for the binary 0b110100110001 is '0x', so carry on.
```

Comment:
We still have a problem,

What if a string is 0x123abc?

It is a valid hex string, but also a valid Base64 encoded string.
Here we will treat it always as hex, but input may have been originally base64 encoded.

Ideally the val.(type) should specify what type of string it is. Otherwise we are screwed.

Reply:
Yes, this is still a problem that needs a long term solution. :+1: 
However, in the case of hashes this should be super rare, since it would have to randomly have both the prefix and no illegal hex characters throughout, and that probability drops with longer strings.

Reply:
Seems like there are deeper problems here, I created this ticket to track a long term solution:

https://app.shortcut.com/chainlinklabs/story/28356/fix-binary-parsing-code-in-the-pipeline

Reply:
interesting